runtime/src/cost_tracker.rs

```diff
@@ -112,12 +134,22 @@ impl CostTracker {
         (costliest_account, costliest_account_cost)
     }
 
-    fn would_fit(&self, keys: &[Pubkey], cost: &u64) -> Result<(), CostTrackerError> {
+    fn would_fit(
+        &self,
+        keys: &[Pubkey],
+        cost: &u64,
+        is_vote: bool,
+    ) -> Result<(), CostTrackerError> {
         // check against the total package cost
         if self.block_cost + cost > self.block_cost_limit {
             return Err(CostTrackerError::WouldExceedBlockMaxLimit);
         }
 
+        // if vote transaction, check if it exceeds vote_transaction_limit
+        if is_vote && self.vote_cost + cost > self.vote_cost_limit {
```

Comment:
```suggestion
        if is_vote && self.vote_cost.saturating_add(cost) > self.vote_cost_limit {
```

Reply:
updated all naked `+` to `saturating_add`